kube-app-testing.sh

```diff
@@ -520,106 +432,105 @@ update_aws_sec_group () {
   fi
 }
 
-
 create_gs_cluster () {
   info "Creating new tenant cluster."
 
   # create a new cluster
-  if ! CLUSTER_DETAILS=$(curl ${GS_API_URL}/v5/clusters/ -X POST \
-      -H "Content-Type: application/json" \
-      -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-      -d "$(gen_gs_blob cluster)") ; then
+  if ! CLUSTER_DETAILS="$(gsctl create cluster --output=json --file - <<EOF
+api_version: v5
+owner: conformance-testing
+release_version: ${GS_RELEASE}
+name: ${CLUSTER_NAME}
+master_nodes:
+  high_availability: false
+labels:
+  circleci-branch: "${CIRCLE_BRANCH:-no}"
+  circleci-build-num: "${CIRCLE_BUILD_NUM:-no}"
+  github-repo: "${CIRCLE_PROJECT_REPONAME:-no}"
+  github-user: "${CIRCLE_PROJECT_USERNAME:-no}"
+  owner: "ci"
+nodepools:
+- availability_zones:
+    zones:
+    - "${AVAILABILITY_ZONE}"
+  scaling:
+    min: ${SCALING_MIN}
+    max: ${SCALING_MAX}
+  node_spec:
+    aws:
+      instance_type: m5.xlarge
+      use_alike_instance_types: true
+EOF
+)" ; then
     err "Cluster creation failed."
     exit 3
   fi
 
   CLUSTER_ID=$(jq -r .id <<< "${CLUSTER_DETAILS}")
 
-  # make sure we're not too fast for the API
-  sleep 5
-
   # write cluster details to file to run a manual cleanup later if required.
   echo "export CLUSTER_ID=${CLUSTER_ID}" > ${ENV_DETAILS_FILE}
   echo "export CLUSTER_TYPE=${CLUSTER_TYPE}" >> ${ENV_DETAILS_FILE}
-  echo "export GS_API_URL=${GS_API_URL}" >> ${ENV_DETAILS_FILE}
   if [ $KEEP_AFTER_TEST ]; then
     echo "export KEEP_AFTER_TEST=${KEEP_AFTER_TEST}" >> ${ENV_DETAILS_FILE}
   fi
 
-  info "Creating nodepool for cluster ${CLUSTER_ID}"
-  # create a nodepool
-  curl ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/nodepools/ -X POST \
-    -H "Content-Type: application/json" \
-    -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-    -d "$(gen_gs_blob nodepool)"
-
-  if [[ "$?" -gt 0 ]]; then
-    err "Nodepool creation failed."
-    exit 3
+  # align config with kind clusters
+  if [[ ! -d ${CONFIG_DIR} ]]; then
+    mkdir ${CONFIG_DIR}
   fi
 
-  info "Adding labels to cluster ${CLUSTER_ID}"
-  # label the cluster with some useful information. failure to label a cluster
-  # doesn't cause a job failure
-  if ! curl ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/labels/ -X PUT \
-      -H "Content-Type: application/json" \
-      -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-      -d "$(gen_gs_blob addlabels)" ; then
-    err "Could not label cluster, however the job will continue."
-  fi
+  info "Sleeping 10 seconds before create kubeconfig attempt"
+  sleep 10
```

Comment:
Is it a race condition? If yes, you should probably re-run the test after timeout.

Reply:
The actual action is wrapped in a loop to make sure it is retried. So IMO this should be good.